packages/curriculum-compiler-json/compilers/glossary.js

```diff
@@ -0,0 +1,22 @@
+const countNodes = require('unist-util-size');
+const { compileNodeToMarkdown } = require('./helpers');
+
+const isHeading1 = (node) => node.type === 'heading' && node.depth === 1;
+
+module.exports = function glossary(ast) {
+  const heading1Count = countNodes(ast, isHeading1);
+
+  if (heading1Count !== 1 || !isHeading1(ast.children[0])) {
+    throw Error('Glossary entry MUST contain a single H1 as title');
```

Comment:
Maybe also include that it must be the first child?

```suggestion
    throw Error('Glossary entry MUST contain a single H1 at the top');
```